ynr/apps/sopn_parsing/helpers/extract_pages.py

```diff
@@ -18,6 +21,88 @@ def extract_pages_for_ballot(ballot, manual_upload=False):
     :type ballot: candidates.models.Ballot
 
     """
+    if ballot.sopn.uploaded_file.path.endswith(".pdf"):
+        pass
+    elif ballot.sopn.uploaded_file.path.endswith(".html"):
+        print(
+            f"Warning! HTML file types are not supported; trying to convert {ballot.sopn.uploaded_file.path} to pdf"
+        )
+        html_file_path = ballot.sopn.uploaded_file.path
+        html_file_name = ballot.sopn.uploaded_file.name
+        pdf_file_path = html_file_path + ".pdf"
```

Comment:
One thing I am a little unsure about is how this will work in production. As when you take the path of the original file in these lines:
```
        html_file_path = ballot.sopn.uploaded_file.path
         html_file_name = ballot.sopn.uploaded_file.name
         pdf_file_path = html_file_path + ".pdf"
```
In production the path will be to the S3 bucket, and I am not sure if pypandoc will be able to write directly to it. It might be better to set the output as a [tempfile](https://docs.python.org/3/library/tempfile.html) which you can then use to read the content from below. As when the official document is saved the PDF will be stored in the S3 bucket as normal.

Reply:
Really good point. Will do, thanks.